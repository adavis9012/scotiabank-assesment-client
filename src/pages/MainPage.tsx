import React from "react";
import Login from "../components/organisms/Login";
import './styles/MainPage.scss';

const MainPage = ((props: any) => {
    return (
        <div className="mainPage">
            <section className="mainPage_content">
                <h1>Prueba Tecnica Scotiabank</h1>
                <p>
                    Sitio web desarrollado por <strong>David Velandia</strong>
                </p>
                <p>
                    Para empezar ingrese utilizando las credenciales enviadas por correo.
                </p>
                <p>
                    Tambien puede crear un nuevo usuario, sin embargo, este no tendra ninguna cuenta ni movimientos.
                </p>
                <hr/>
                <h2>Github</h2>
                <p>GitHub Personal <a href="https://github/adavis9012">adavis9012</a></p>
                <h2>Linkedin</h2>
                <a href="https://www.linkedin.com/in/andres-david-velandia">https://www.linkedin.com/in/andres-david-velandia</a>
                <br/>
                <br/>
                <hr/>
                <h2>Gracias!</h2>
            </section>
            <aside className="mainPage_login">
                <Login {...props} />
            </aside>
        </div>
    );
});

export default MainPage;
