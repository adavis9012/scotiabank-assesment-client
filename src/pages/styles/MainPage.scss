@import "../../settings/variables";
@import "../../settings/mixins";

.mainPage {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column-reverse;

  @include media(tablet, desktop) {
    flex-wrap: nowrap;
    flex-direction: row;
  }

  &_content {
    width: 100%;
    padding: 1rem;

    @include media(tablet, desktop) {
      width: 70%;
    }
  }

  &_login{
    width: 94%;
    padding: 3%;

    @include media(tablet, desktop) {
      width: 30%;
      padding: 1rem;
    }
  }
}