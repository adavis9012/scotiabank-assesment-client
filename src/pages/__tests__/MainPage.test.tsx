import React from 'react';
import { shallow } from 'enzyme';
import MainPage from '../MainPage';

describe('<MainPage />', () => {
    const localStorageMock = {
        getItem: jest.fn(),
        setItem: jest.fn(),
        clear: jest.fn(),
        removeItem: jest.fn(),
    };

    describe('should match snapshot', () => {

        beforeEach(() => {
            Object.defineProperty(window, 'localStorage', {
                value: localStorageMock
            });
        });

        it('when no is not logged', () => {
            const component = shallow(<MainPage />);

            expect(component).toMatchSnapshot();
            expect(component.find('Login')).toHaveLength(1);
            expect(component.find('Logout')).toHaveLength(0);
        });

        it('when is logged', () => {
            localStorageMock.getItem = jest.fn((e) => 'token');

            const component = shallow(<MainPage />);

            expect(component).toMatchSnapshot();
            expect(component.find('Login')).toHaveLength(0);
            expect(component.find('Logout')).toHaveLength(1);
        });
    });

    it('should clear storage when handler is invoked',() => {
        localStorageMock.getItem = jest.fn((e) => 'token');

        Object.defineProperty(window, 'localStorage', {
            value: localStorageMock
        });

        const component = shallow(<MainPage />);
        const LogoutProps:any = component.find('Logout').props();

        LogoutProps.handleLogin();
        expect(localStorageMock.removeItem).toBeCalled();
    });
});
