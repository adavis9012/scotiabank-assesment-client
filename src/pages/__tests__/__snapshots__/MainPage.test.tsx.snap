// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> should match snapshot when is logged 1`] = `
<div
  className="mainPage"
>
  <section
    className="mainPage_content"
  >
    <h1>
      Prueba Tecnica Scotiabank
    </h1>
    <p>
      Para empezar ingrese utilizando las credenciales enviadas por correo.
    </p>
    <p>
      Tambien puede crear un nuevo usuario, sin embargo, este no tendra ninguna cuenta ni movimientos.
    </p>
    <hr />
    <p>
      Sitio web desarrollado por 
      <strong>
        David Velandia
      </strong>
    </p>
    <h2>
      Github
    </h2>
    <p>
      <strong>
        GitHub 
      </strong>
       
      <a
        href="https://github.com/adavis9012/scotiabank-assesment-client"
      >
        Cliente
      </a>
    </p>
    <p>
      <strong>
        GitHub 
      </strong>
       
      <a
        href="https://github.com/adavis9012/scotiabank-assesment-server"
      >
        Servidor
      </a>
    </p>
    <p>
      <strong>
        GitHub Personal
      </strong>
       
      <a
        href="https://github/adavis9012"
      >
        adavis9012
      </a>
    </p>
    <h2>
      Linkedin
    </h2>
    <a
      href="https://www.linkedin.com/in/andres-david-velandia"
    >
      https://www.linkedin.com/in/andres-david-velandia
    </a>
    <br />
    <br />
    <hr />
    <h2>
      Gracias!
    </h2>
  </section>
  <aside
    className="mainPage_login"
  >
    <Logout
      handleLogin={[Function]}
    />
  </aside>
</div>
`;

exports[`<MainPage /> should match snapshot when no is not logged 1`] = `
<div
  className="mainPage"
>
  <section
    className="mainPage_content"
  >
    <h1>
      Prueba Tecnica Scotiabank
    </h1>
    <p>
      Para empezar ingrese utilizando las credenciales enviadas por correo.
    </p>
    <p>
      Tambien puede crear un nuevo usuario, sin embargo, este no tendra ninguna cuenta ni movimientos.
    </p>
    <hr />
    <p>
      Sitio web desarrollado por 
      <strong>
        David Velandia
      </strong>
    </p>
    <h2>
      Github
    </h2>
    <p>
      <strong>
        GitHub 
      </strong>
       
      <a
        href="https://github.com/adavis9012/scotiabank-assesment-client"
      >
        Cliente
      </a>
    </p>
    <p>
      <strong>
        GitHub 
      </strong>
       
      <a
        href="https://github.com/adavis9012/scotiabank-assesment-server"
      >
        Servidor
      </a>
    </p>
    <p>
      <strong>
        GitHub Personal
      </strong>
       
      <a
        href="https://github/adavis9012"
      >
        adavis9012
      </a>
    </p>
    <h2>
      Linkedin
    </h2>
    <a
      href="https://www.linkedin.com/in/andres-david-velandia"
    >
      https://www.linkedin.com/in/andres-david-velandia
    </a>
    <br />
    <br />
    <hr />
    <h2>
      Gracias!
    </h2>
  </section>
  <aside
    className="mainPage_login"
  >
    <Login />
  </aside>
</div>
`;
